src/material-experimental/mdc-slider/slider.e2e.spec.ts

```diff
@@ -6,11 +6,133 @@
+    beforeAll(() => { slider = getRangeSlider(); });
+
+    it('should update the start thumb value on slide', async () => {
+      await slideToValue(slider, 35, Thumb.START);
+      expect(await getSliderValue(slider, Thumb.START)).toBe(35);
+    });
+
+    it('should update the end thumb value on slide', async () => {
+      console.log('value:', await getSliderValue(slider, Thumb.END));
+      await slideToValue(slider, 55, Thumb.END);
+      console.log('value:', await getSliderValue(slider, Thumb.END));
+      expect(await getSliderValue(slider, Thumb.END)).toBe(55);
+    });
+
+    it('should update the start thumb value on click between thumbs '
+      + 'but closer to the start thumb', async () => {
+        await setValueByClick(slider, 49);
+        expect(await getSliderValue(slider, Thumb.START)).toBe(49);
+        expect(await getSliderValue(slider, Thumb.END)).toBe(100);
+    });
+
+    it('should update the end thumb value on click between thumbs '
+      + 'but closer to the end thumb', async () => {
+        await setValueByClick(slider, 51);
+        expect(await getSliderValue(slider, Thumb.START)).toBe(0);
+        expect(await getSliderValue(slider, Thumb.END)).toBe(51);
+    });
+  });
 });
+
+/** Returns the current value of the slider. */
+async function getSliderValue(slider: ElementFinder, thumbPosition: Thumb): Promise<number> {
+  const inputs = await slider.all(by.css('.mdc-slider__input'));
+  return thumbPosition === Thumb.END
+    ? Number(await inputs[inputs.length - 1].getAttribute('value'))
+    : Number(await inputs[0].getAttribute('value'));
+}
+
+/** Clicks on the MatSlider at the coordinates corresponding to the given value. */
+async function setValueByClick(slider: ElementFinder, value: number): Promise<void> {
+  return clickElementAtPoint(slider, await getCoordsForValue(slider, value));
+}
+
+/** Clicks on the MatSlider at the coordinates corresponding to the given value. */
+async function slideToValue
+  (slider: ElementFinder, value: number, thumbPosition: Thumb): Promise<void> {
+    const webElement = await getElement(slider).getWebElement();
+    const startCoords = await getCoordsForValue(
+      slider,
+      await getSliderValue(slider, thumbPosition),
+    );
+    const endCoords = await getCoordsForValue(slider, value);
+    return await browser.actions()
+      .mouseMove(webElement, startCoords)
+      .mouseDown()
+      .mouseMove(webElement, endCoords)
+      .mouseUp()
+      .perform();
+}
+
+/** Returns the x and y coordinates for the given slider value. */
+async function getCoordsForValue(slider: ElementFinder, value: number): Promise<Point> {
+  const inputs = await slider.all(by.css('.mdc-slider__input'));
+
+  const min = Number(await inputs[0].getAttribute('min'));
+  const max = Number(await inputs[inputs.length - 1].getAttribute('max'));
+  const percent = (value - min) / (max - min);
+
+  const {width, height} = await getBoundingClientRect(slider);
```

Comment:
Protractor has a `getSize` method so you don't have to delegate to `getBoundingClientRect`.